src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java

```diff
@@ -24,15 +26,52 @@ public class ProgrammingExerciseTaskService {
     private final ExerciseHintRepository exerciseHintRepository;
 
     /**
-     * Pattern that is used to extract the tasks (capturing group "name") and test case names (capturing group "tests") from the problem statement.
-     * Example: "[task][Implement BubbleSort](testBubbleSort,testBubbleSortHidden)". Following values are extracted by the named capturing groups:
-     * - name: "Implement BubbleSort"
-     * - tests: "testBubbleSort,testBubbleSortHidden"
-     *
-     * This is coupled to the value used in `ProgrammingExerciseTaskExtensionWrapper` and `TaskCommand` in the client
+     * Pattern that is used to extract the tasks (capturing group {@code name}) and test case names (capturing groups {@code tests}) from the problem statement.
+     * Example: "[task][Implement BubbleSort](testBubbleSort,testBubbleSortHidden)". Following groups are extracted by the capturing groups:
+     * <ul>
+     * <li>name: {@code Implement BubbleSort}
+     * <li>tests: {@code testBubbleSort,testBubbleSortHidden}
+     * </ul>
+     * <p>
+     * The first section - withing square brackets - captures the task identifier {@code [task]}.<br>
+     * The second section {@code name} - within square brackets - matches the task name, allowing any characters but square brackets.<br>
+     * The third and last section {@code tests} - within round brackets - captures zero, one or multiple test cases. Multiple test cases get separated by a comma and optional
+     * whitespace. Each test case contains a test name with or without round brackets. These round brackets may contain method parameters. Test names may contain any characters but
+     * round brackets, whitespace or commas. Method parameters only exclude round brackets. After round brackets a test case may contain additional characters excluding round
+     * brackets or commas.<br>
+     * Therefore, allowed test names are among others {@code testName}, {@code testName()}, {@code testName(1234, 12)}, {@code testName(testValue)[1]}, {@code Test Name}.<br>
+     * For multiple testcases it's {@code testName,otherTestName()} or {@code testName,     otherTestName()}.<br>
+     * <p>
+     * This is coupled to the value used in `ProgrammingExerciseTaskExtensionWrapper`, `ProgrammingExerciseInstructionAnalysisService`, and `TaskCommand` in the client
+     * If you change the regex, make sure to change it in all places!
```

Comment:
Can you reference the other places?

Reply:
What do you mean by this? They are directly mentioned above in the text.

Reply:
Sorry, I overlooked that this was the case